### PR TITLE
feat: add the muttrc parser

### PR DESCRIPTION
**Pull request by Freed-Wu**
Screenshot from 2024-02-14 09-50-29

The above is tree-sitter, the below is vim syn.

### CODE REVIEW

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -47,6 +47,7 @@ for ft, lang in pairs {
   xsd = "xml",
   xslt = "xml",
   sbt = "scala",
+  neomuttrc = "muttrc",
```
Comment (ObserverOfTime) [inline lua/nvim-treesitter/parsers.lua:50]:
```suggestion
```

File: queries/muttrc/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+] @variable.builtin
+
+(command) @keyword
+
+(source_directive
+  (command) @keyword.import)
+
+(uri) @string.special.url
+
+(key_name) @type.builtin
```
Comment (ObserverOfTime) [inline queries/muttrc/highlights.scm:35]:
```suggestion
(key_name) @constant.builtin
```

File: queries/muttrc/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  ">"
+] @punctuation.bracket
+
+"," @punctuation.delimiter
+
+[
+  "&"
+  "?"
+  "*"
+] @punctuation.special
```
Comment (ObserverOfTime) [inline queries/muttrc/highlights.scm:53]:
```suggestion
] @character.special
```

Comment (Freed-Wu) [reply queries/muttrc/highlights.scm]: Fixed.

File: queries/muttrc/highlights.scm (hunk 2 above)
Comment (Freed-Wu) [reply queries/muttrc/highlights.scm]: Fixed.

File: lua/nvim-treesitter/parsers.lua (hunk 1 above)
Comment (Freed-Wu) [reply lua/nvim-treesitter/parsers.lua:50]: @flatcap (the author of neomuttrc.vim) what is the difference between muttrc and neomuttrc? IMO, the latter should have more customized options?

Comment (ObserverOfTime) [reply lua/nvim-treesitter/parsers.lua:50]: Oh, it is in Vim. Nevermind.

**Review by ObserverOfTime — APPROVED** [review]

**Pull request merged**